lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -1114,44 +1145,74 @@ def _get_app_or_unit(self) -> Union[Application, Unit]:
     @property
     def private_key(self) -> Optional[PrivateKey]:
         """Return the private key."""
+        if self._private_key:
+            return self._private_key
         if not self._private_key_generated():
             return None
         secret = self.charm.model.get_secret(label=self._get_private_key_secret_label())
         private_key = secret.get_content(refresh=True)["private-key"]
         return PrivateKey.from_string(private_key)
 
-    def _generate_private_key(self) -> None:
+    def _ensure_private_key(self) -> None:
```

Comment:
Some docstring would be useful to understand what this function does

Reply:
Added